src/datastore/src/userstore.js

```diff
@@ -33,6 +35,54 @@ class UserStore extends NetworkStore {
   }
 
   /**
+   * Find all users. A query can be optionally provided to return
+   * a subset of all users or omitted to return all users.
+   *
+   * @param {Query} [query] Query used to filter entities.
+   * @param {Object} [options] Options
+   * @param {Boolean} [options.discover] Discover users.
+   * @return {Observable} Observable.
+   */
+  find(query, options = {}) {
+    // If `options.discover`, use
+    // [User Discovery](http://devcenter.kinvey.com/guides/users#lookup)
+    // instead of querying the user namespace directly.
+    if (options.discover === true) {
```

Comment:
If you agree that separating the two methods them makes sense, let's do it. I think it reduces confusion for the end user, and it's better to be consistent with other v3 SDKs as compared to JS v1.

The unit test on `find()` was to clearly define what the expected behavior is (if we say `find` is a supported method on the `UserStore`). I suspect there's at least some difference in how it behaves on users vs data, but I'm not sure either.